src/service-worker/ServiceWorker.ts

```diff
@@ -488,28 +489,30 @@ export class ServiceWorker {
       vibrate: notification.vibrate
     };
 
-    notificationOptions = ServiceWorker.filterNotificationOptions(notificationOptions, persistNotification === 'force');
+    notificationOptions = ServiceWorker.fixPlatformSpecificDisplayIssues(notificationOptions);
     return self.registration.showNotification(notification.heading, notificationOptions);
   }
 
-  static filterNotificationOptions(options: any, forcePersistNotifications?: boolean): any {
-    /**
-     * Due to Chrome 59+ notifications on Mac OS X using the native toast style
-     * which limits the number of characters available to display the subdomain
-     * to 14 with requireInteraction and 28 without, we force Mac OS X Chrome
-     * notifications to be transient.
-     */
-    const clone = { ...options };
+  /**
+   * Fixes display issue with some notification options causing the notification to never show!
+   * This happens when setting requireInteraction = true on the following platforms;
+   *   * macOS 10.15+ - Chrome based browsers
+   *      - https://bugs.chromium.org/p/chromium/issues/detail?id=1007418
+   *   * macOS 10.14+ - Opera
+   *      - https://forums.opera.com/topic/31334/push-notifications-with-requireinteraction-true-do-not-display-on-macos
+   * @param notificationOptions - Value passed to ServiceWorkerRegistration.prototype.showNotification
+   */
+  static fixPlatformSpecificDisplayIssues(notificationOptions: any): any {
+    const clone = { ...notificationOptions };
     const browser = OneSignalUtils.redetectBrowserUserAgent();
 
-    if (browser.chrome &&
-      browser.mac &&
-      clone) {
+    if (browser.chrome && browser.mac && Utils.isVersionAtLeast(browser.osversion, 10.15)) {
```

Comment:
it looks like before we were (trying) to set requireInteraction to false for all mac chrome browsers but now we are only doing it for 10.15+ 

Is this comment from the old code no longer true?

> * Due to Chrome 59+ notifications on Mac OS X using the native toast style
> * which limits the number of characters available to display the subdomain	
> * to 14 with requireInteraction and 28 without, we force Mac OS X Chrome
> * notifications to be transient.

Reply:
This was removed intentionally, it was always 'force' or false. This logic was broken for a long time so this is now the expected behavior. Also can't enable anyway for macOS 10.15 if / util Chrome or macOS fixes this issue